tiledb/sm/array_schema/array_schema.h

```diff
@@ -249,6 +249,34 @@ class ArraySchema {
    */
   Status drop_attribute(const std::string& attr_name);
 
+  /**
+   * Renames an attribute.
+   *
+   * @param old_name The old attribute name
+   * @param new_name The new attribute name
+   * @param is_new_uri The flag to generate new array schema uri
+   * @return Status
+   */
+  Status rename_attribute(
+      const std::string& old_name,
+      const std::string& new_name,
+      bool is_new_uri);
+
+  /**
+   * Returns a vector of used names of attributes
+   *
+   * @return all used names
+   */
+  std::vector<std::string> attribute_used_names() const;
+
+  /**
+   * Get the current name for a used name.
+   *
+   * @param used_name The used name
+   * @return current name
+   */
+  std::string attribute_current_name(const std::string& used_name) const;
```

Comment:
I suggest renaming to `attribute_new_name` for consistency everywhere.

Reply:
refactored the function to "Status get_attribute_new_name(const std::string& old_name, std::string* new_name) const". Thanks a lot for all the comments and suggestions.